pkg/common/parameters_test.go

```diff
@@ -199,6 +201,37 @@ func TestExtractAndDefaultParameters(t *testing.T) {
 				Labels:          map[string]string{},
 			},
 		},
+		{
+			name:       "storage pool parameters",
+			parameters: map[string]string{ParameterKeyType: "hyperdisk-balanced", ParameterKeyStoragePools: "projects/my-project/zones/us-central1-a/storagePools/storagePool-1,projects/my-project/zones/us-central1-b/storagePools/storagePool-2"},
+			labels:     map[string]string{},
+			expectParams: DiskParameters{
+				DiskType:        "hyperdisk-balanced",
+				ReplicationType: "none",
+				Tags:            map[string]string{},
+				Labels:          map[string]string{},
+				StoragePools: []StoragePool{
+					{
+						Project:      "my-project",
+						Zone:         "us-central1-a",
+						Name:         "storagePool-1",
+						ResourceName: "projects/my-project/zones/us-central1-a/storagePools/storagePool-1",
+					},
+					{
+						Project:      "my-project",
+						Zone:         "us-central1-b",
+						Name:         "storagePool-2",
+						ResourceName: "projects/my-project/zones/us-central1-b/storagePools/storagePool-2",
+					},
+				},
+			},
+		},
+		{
+			name:       "invalid storage pool parameters",
+			parameters: map[string]string{ParameterKeyType: "hyperdisk-balanced", ParameterKeyStoragePools: "zones/us-central1-a/storagePools/storagePool-1"},
+			labels:     map[string]string{},
+			expectErr:  true,
+		},
```

Comment:
Add a test that tries to break the regular expression, eg /zones/zones and stuff like that.

Reply:
Added.